deprecations.md

```diff
@@ -0,0 +1,48 @@
+# Deprecations
```

Comment:
I like this section, however don't we want to put it to changelog? These are essentially upcoming changes and we want everybody to see it, e.g. when checking changelog. It would be similar to our release notes where we have "coming soon" section on top.

Reply:
I was initially against it (maybe link it?), but now it's growing on me. Will try to see how it looks.